include/deal.II/grid/grid_tools.h

```diff
@@ -157,30 +157,18 @@ namespace GridTools
    * get the same result using <code>cell-@>measure()</code>, but this
    * function also works for cells that do not exist except that you make it
    * up by naming its vertices from the list.
-   *
-   * @note Since <code>dim</code> does not explicitly appear in the function
-   * signature, you must explicitly specify both dimensions when using this
-   * function.
    */
-  template <int dim, int spacedim>
-  double cell_measure (const std::vector<Point<spacedim> > &all_vertices,
+  template <int dim>
+  double cell_measure (const std::vector<Point<dim> > &all_vertices,
                        const unsigned int (&vertex_indices)[GeometryInfo<dim>::vertices_per_cell]);
 
-
   /**
-   * Same as the last function, but for the codimension zero case. This extra
-   * function allows one to write
-   *
-   * @code
-   * cell_measure(vertices, indices);
-   * @endcode
-   *
-   * and have the template function argument (i.e. <code>dim</code>) inferred
-   * automatically.
+   * A version of the last function that can accept input for nonzero
+   * codimension cases. This function only exists to aid generic programming
+   * and calling it will just raise an exception.
    */
-  template <int dim>
-  double cell_measure (const std::vector<Point<dim> > &all_vertices,
-                       const unsigned int (&vertex_indices)[GeometryInfo<dim>::vertices_per_cell]);
+  template <int dim, typename T>
+  double cell_measure (const T &, ...);
```

Comment:
In other words, I think you need to write this as follows:
```
  template <int dim>
  double cell_measure (const std::vector<Point<dim> > &all_vertices,
                                     ...);
```
This way, the `dim` argument is deducible, and the second argument is whatever it is. Of course, you could also write it as
```
  template <int dim, typename T>
  double cell_measure (const std::vector<Point<dim> > &all_vertices,
                                     T &);
```

Here's my little test code:
```
template <int dim> class Point 
{};

template <int dim>
void foo (const Point<dim> &p,
          const double (&array)[dim]);

template <int dim>
void foo (const Point<dim> &p,
          ...);

int main ()
{
  Point<3> p;
  double array[444];  // mismatched array
  foo (p, array);
}
```
If you compile it, you get this:
```
tmp/g> c++ -c x.cc
tmp/g> nm -C x.o
                 U void foo<3>(Point<3> const&, ...)
0000000000000000 T main
```
which calls the correct function.